components/common/dropdown/Dropdown.tsx

```diff
@@ -0,0 +1,211 @@
+   */
+  dropdownClassName?: string;
+
+  /**
+   * 옵션 버튼에 적용할 커스텀 스타일
+   * @example 'text-blue-700 hover:bg-blue-200'
+   */
+  optionClassName?: string;
+};
+
+/**
+ * ## 드롭다운 로직
+ * 1. **드롭다운을 누르기 전**:
+ *   - 각 타입에 맞게 `trigger`나 `initialLabel`을 표시
+ *
+ * 2. **드롭다운을 누른 후**:
+ *   - `options`로 전달한 값들을 표시
+ *   - `width`나 `dropdownClassName`, `optionClassName`을 적용할 경우 커스텀 스타일이 적용
+ *
+ * 3. **각 옵션을 눌렀을 때**:
+ *   - `handleSelect` 함수가 실행되며 `onSelect` prop으로 넘겨준 콜백이 호출
+ *   - **action 타입**: 드롭다운이 닫히며 추가 동작(이동, 모달 열기 등)이 발생
+ *   - **select 타입**: 선택한 값이 드롭다운에 표시되며 드롭다운이 닫힘
+ * 
+ * @example   const options1 = [
+  { label: 'Red', value: 'red' },
+  { label: 'White', value: 'white' },
+  { label: 'Sparkling', value: 'sparkling' },
+];
+
+const options2 = [
+  { label: '수정하기', value: 'edit' },
+  { label: '삭제하기', value: 'delete' },
+];
+
+const handleSelect = (value: string | number) => {
+  console.log(`Selected value: ${value}`);
+  // 여기에 원하는 액션을 추가
+};
+
+<Dropdown
+        options={options1}
+        onSelect={handleSelect}
+        type="select"
+        width={368} // 커스텀 너비
+        initialLabel="Red" // 초기 레이블 설정
+      />
+
+       <Dropdown
+        options={options2}
+        onSelect={handleSelect}
+        type="action"
+        trigger={<span className="cursor-pointer">O</span>}
+        optionClassName="text-center" // 옵션 항목의 스타일
+      />
+ */
+function Dropdown({
+  options,
+  onSelect,
+  type,
+  trigger,
+  initialLabel,
+  width = 126,
+  dropdownClassName,
+  optionClassName,
+}: DropdownProps) {
+  const [isOpen, setIsOpen] = useState(false);
+  const [selectedLabel, setSelectedLabel] = useState<string | undefined>(
+    initialLabel
+  );
+  const dropdownRef = useRef<HTMLDivElement | null>(null);
+
+  // 클릭 외부 감지
+  useEffect(() => {
+    const handleClickOutside = (event: MouseEvent) => {
+      if (
+        dropdownRef.current &&
+        !dropdownRef.current.contains(event.target as Node)
+      ) {
+        setIsOpen(false);
+      }
+    };
+
+    // 전역 클릭 이벤트 리스너 등록
+    document.addEventListener('mousedown', handleClickOutside);
+
+    // 컴포넌트 언마운트 시 리스너 제거
+    return () => {
+      document.removeEventListener('mousedown', handleClickOutside);
+    };
+  }, []);
+
+  const handleSelect = (option: Option) => {
+    onSelect(option.value);
+    setSelectedLabel(option.label);
+    setIsOpen(false);
+  };
+
+  return (
+    <div className="relative inline-block text-left" ref={dropdownRef}>
+      {type === 'action' ? (
+        <div onClick={() => setIsOpen(!isOpen)}>{trigger}</div>
+      ) : (
+        <div
+          onClick={() => setIsOpen(!isOpen)}
+          className={`flex items-center cursor-pointer border rounded-[16px] px-4 py-2 ${dropdownClassName} text-gray-700 font-medium bg-white ring-1 ring-gray-300`}
+          style={{ width: `${width}px` }}
+        >
+          <span className={`flex-1 ${isOpen ? 'text-gray-300' : ''}`}>
+            {selectedLabel || initialLabel || 'Select'}
```

Comment:
selectedLabel에서 initialLabel을 초기값으로 받고 있으니 
```suggestion
            {selectedLabel || 'Select'}
```
이렇게 해도 괜찮을거 같아요.